src/Plugin/EntityReferenceSelection/OgUserSelection.php

```diff
@@ -153,11 +153,13 @@ public function entityQueryAlter(SelectInterface $query) {
     }
 
     // Left join to the OG membership base table.
-    $query->leftJoin('og_membership', 'ogm', "base_table.uid = ogm.uid");
+    $query->leftJoin('og_membership', 'ogm', "base_table.uid = ogm.uid AND ogm.entity_type != :entity_type AND ogm.entity_id != :entity_id", [
+      ':entity_type' => $group->getEntityTypeId(),
+      ':entity_id' => $group->id(),
+    ]);
 
     // Exclude any users who are in the current group.
-    $query->condition('ogm.entity_type', $group->getEntityTypeId(), '!=');
-    $query->condition('ogm.entity_id', $group->id(), '!=');
+    $query->condition('ogm.id', NULL, 'IS');
```

Comment:
That doesn't look like the right way to do that. The core Condition class has an isNull() method, which is probably the right thing to use.